contracts/protocol/facets/SellerHandlerFacet.sol

```diff
@@ -94,44 +94,176 @@ contract SellerHandlerFacet is SellerBase {
+        }
+
+        if (seller.treasury != _seller.treasury) {
+            require(_seller.treasury != address(0), INVALID_ADDRESS);
+            // Update treasury
+            seller.treasury = _seller.treasury;
+        }
+
+        if (_seller.operator != seller.operator) {
+            require(_seller.operator != address(0), INVALID_ADDRESS);
+            // Operator address owner must approve the update for prevent front-running
+            lookups.sellerPendingUpdates[seller.id].operator = _seller.operator;
+            needsApproval = true;
+        }
+
+        if (_seller.clerk != seller.clerk) {
+            require(_seller.clerk != address(0), INVALID_ADDRESS);
+            // Clerk address owner must approve the update for prevent front-running
+            lookups.sellerPendingUpdates[seller.id].clerk = _seller.clerk;
+            needsApproval = true;
+        }
+
+        if (needsApproval) {
+            lookups.sellerPendingUpdates[seller.id].id = _seller.id;
+            emit SellerUpdateRolesRequested(_seller.id, _seller, sender);
+        }
+
+        // Notify watchers of state change
+        emit SellerUpdated(_seller.id, seller, _authToken, sender);
+    }
+
+    function approveSellerUpdate(uint256 _sellerId) external {
+        // Cache protocol lookups for reference
+        ProtocolLib.ProtocolLookups storage lookups = protocolLookups();
+        address sender = msgSender();
+
+        // Get seller pending update
+        Seller storage _sellerPendingUpdate = lookups.sellerPendingUpdates[_sellerId];
+
+        require(
+            _sellerPendingUpdate.admin != address(0) ||
+                _sellerPendingUpdate.operator != address(0) ||
+                _sellerPendingUpdate.clerk != address(0),
+            NO_PENDING_SELLER_ROLE_UPDATE
+        );
+
+        sellerPreUpdateChecks(_sellerPendingUpdate);
+
+        // Get storage location for seller
+        (, Seller storage seller, ) = fetchSeller(_sellerId);
+
+        // Approve operator address
+        if (_sellerPendingUpdate.operator == sender) {
+            // If operator changed, update the operator and transfer the ownership of NFT voucher
+            if (seller.operator != _sellerPendingUpdate.operator) {
+                delete lookups.sellerIdByOperator[seller.operator];
+
+                // Update operator
+                seller.operator = _sellerPendingUpdate.operator;
+
+                // Transfer ownership of NFT voucher to new operator
+                IBosonVoucher(lookups.cloneAddress[seller.id]).transferOwnership(seller.operator);
+
+                // Store new seller id by operator mapping
+                lookups.sellerIdByOperator[sender] = _sellerId;
+            }
+
+            // Delete pending update operator
+            _sellerPendingUpdate.operator = address(0);
+        }
+
+        // Approve admin address
+        if (_sellerPendingUpdate.admin == sender) {
+            if (seller.admin != _sellerPendingUpdate.admin) {
```

Comment:
not sure if I understood. If we change for `_sellerPendingUpdate.admin != address(0)` this will be a different validation. We can't remove `_sellerPendingUpdate.admin == sender` because this is what check if the caller is the correct address, and why update even when the current admin is equal to new admin?

Reply:
Yes, what I wrote wasn't accurate. What I meant was, can it really happen that `_sellerPendingUpdate.admin` and `seller.admin` are the same?

I thought that [this check](https://github.com/bosonprotocol/boson-protocol-contracts/blob/96693a2c6ba18285c0b3296e8631d850cba05830/contracts/protocol/facets/SellerHandlerFacet.sol#L133)
prevents it.

Reply:
make sense, fixed